src/mechanics/cell/Cell.h

```diff
@@ -58,6 +58,35 @@ class Cell : public CellInterface
         return mId;
     }
 
+    Eigen::VectorXd Interpolate(Eigen::VectorXd naturalCoords) const override
+    {
+        const auto& coordinateElement = mElements.CoordinateElement();
+        return coordinateElement.GetNMatrix(naturalCoords) * coordinateElement.ExtractNodeValues();
+    }
+
+    Eigen::VectorXd Interpolate(Eigen::VectorXd naturalCoords, DofType dof) const override
+    {
+        const auto& element = mElements.DofElement(dof);
+        return element.GetNMatrix(naturalCoords) * element.ExtractNodeValues();
+    }
+
+    std::vector<Eigen::VectorXd>
```

Comment:
Should we introduce a custom datatype for that? Maybe for now, just a file `IpData.h` containing `using IpData = Eigen::VectorXd`. 
I used that e.g. for the `DofVector`. First, it was just a typedef. When I needed more functionality, it became a class. But I only needed to change code inside DofVector file, no API changes.

Reply:
Personally, I'm not a big fan of that. At the locations where it is used, I'll know immediately what the type does and what it's member functions are when I read `Eigen::VectorXd`. With `IpData`, I'll have to remember that or look it up, putting undue strain on my small brain :)